Revit_Engine/Query/RevitParametersToPushDifferences.cs

```diff
@@ -0,0 +1,115 @@
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace BH.Engine.Adapters.Revit
+{
+    public static partial class Query
+    {
+        [Description("Returns Property differences between RevitParametersToPush owned by the two input objects." +
+            "This method can be added as Func delegate to the DiffingConfig; if so, it is automatically triggered when Diffing the two objects.")]
+        [Input("obj1", "Past object being compared.")]
+        [Input("obj2", "Following object being compared.")]
+        [Input("comparisonConfig", "Comparison Config to be used during comparison.")]
+        [Output("parametersDifferences", "Differences in terms of RevitParametersToPush found on the two input objects.")]
+        public static List<PropertyDifference> RevitParametersToPushDifferences(this object obj1, object obj2, BaseComparisonConfig comparisonConfig)
+        {
+            List<PropertyDifference> result = new List<PropertyDifference>();
+
+            Dictionary<string, RevitParameter> allParamsDict_obj1 = (obj1 as IBHoMObject)?.GetParametersToPush();
+            Dictionary<string, RevitParameter> allParamsDict_obj2 = (obj2 as IBHoMObject)?.GetParametersToPush();
+
+            if ((!allParamsDict_obj1?.Any() ?? true) || (!allParamsDict_obj2?.Any() ?? true))
+                return result; // no pulled parameters found.
+         
+            List<string> overlappingParams = allParamsDict_obj1.Values.Select(p => p.Name).Intersect(allParamsDict_obj2.Values.Select(p => p.Name)).ToList();
+
+            foreach (string paramName in overlappingParams)
+            {
+                KeyValuePair<string, RevitParameter> overlappingParamTuple_1 = allParamsDict_obj1.First(p => p.Value.Name == paramName);
+                KeyValuePair<string, RevitParameter> overlappingParamTuple_2 = allParamsDict_obj2.First(p => p.Value.Name == paramName);
+
+                RevitParameter parameter1 = overlappingParamTuple_1.Value;
+                RevitParameter parameter2 = overlappingParamTuple_2.Value;
+                string differenceFullName = overlappingParamTuple_2.Key;
+
+                if (parameter1.Value?.Equals(parameter2?.Value) ?? false)
+                    continue; // parameters are equal in value. Continue.
+
+                // Check if we have a RevitComparisonConfig input.
+                RevitComparisonConfig rcc = comparisonConfig as RevitComparisonConfig;
+
+                // Check if effectively the parameter difference is to be considered.
+                if (!Query.ComparisonInclusion(parameter1, parameter2, differenceFullName, rcc).Include)
+                    continue;
+
+                // If we got here, the two parameters are different, and we want to record this difference.
+                PropertyDifference parameterDiff = new PropertyDifference();
+                parameterDiff.DisplayName = paramName + " (RevitParametersToPush)";
```

Comment:
Could we change the suffix to `RevitParameter` in order to stick to the original concept? I think it is enough information.